codegen/src/lower.rs

```diff
@@ -41,6 +41,70 @@ struct ConvertCtx<'a> {
 
 type OptVarType = Option<(Variable, types::Type)>;
 
+enum IFCond {
+    IntCond(IntCC),
+    FloatCond(FloatCC),
+}
+fn opcode_to_cond(op: Opcode) -> IFCond {
+    use self::Opcode::*;
+    use self::OrderedType::*;
+    use self::IFCond::*;
+    match op {
+        Ord(Ordered(t)) => {
+            assert_eq!(t.is_float(), true);
```

Comment:
I think it is safe to write:

```rust
assert!(t.is_float());
```

instead of `assert_eq!`.

Reply:
Thanks! I will fix it next week-end.

If you have problem with some of the comments or the lack of comments, or think some descriptions are not clear, feel free to report them as well. I am trying to document as much as possible from the beginning and make these comments as readable/understandable as possible.

Reply:
Everything is really great so far, you're doing an excellent job!